[ao] Get feature and module names from ModelReportVisualizer class

vspenubarthi opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #81673
* __->__ #81647
* #81589
* #81523

Summary: Added the functionality to be able to get the feature names and
module_fqns from the ModelReportVisualizer class. The purpose of this
addition is so that users can see the exact set of module_fqns or
feature names that they can filter based on, and use this information to
perform their filtering.

Test Plan: python test/test_quantization.py
TestFxModelReportVisualizer.test_get_modules_and_features

Reviewers:

Subscribers:

Tasks:

Tags:

vspenubarthi committed:
[ao] Get feature and module names from ModelReportVisualizer class

Summary: Added the functionality to be able to get the feature names and
module_fqns from the ModelReportVisualizer class. The purpose of this
addition is so that users can see the exact set of module_fqns or
feature names that they can filter based on, and use this information to
perform their filtering.

Test Plan: python test/test_quantization.py
TestFxModelReportVisualizer.test_get_modules_and_features

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vspenubarthi committed:
Update on "[ao] Get feature and module names from ModelReportVisualizer class"


Summary: Added the functionality to be able to get the feature names and
module_fqns from the ModelReportVisualizer class. The purpose of this
addition is so that users can see the exact set of module_fqns or
feature names that they can filter based on, and use this information to
perform their filtering.

Test Plan: python test/test_quantization.py
TestFxModelReportVisualizer.test_get_modules_and_features

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vspenubarthi committed:
Update on "[ao] Get feature and module names from ModelReportVisualizer class"


Summary: Added the functionality to be able to get the feature names and
module_fqns from the ModelReportVisualizer class. The purpose of this
addition is so that users can see the exact set of module_fqns or
feature names that they can filter based on, and use this information to
perform their filtering.

Test Plan: python test/test_quantization.py
TestFxModelReportVisualizer.test_get_modules_and_features

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vspenubarthi committed:
Update on "[ao] Get feature and module names from ModelReportVisualizer class"


Summary: Added the functionality to be able to get the feature names and
module_fqns from the ModelReportVisualizer class. The purpose of this
addition is so that users can see the exact set of module_fqns or
feature names that they can filter based on, and use this information to
perform their filtering.

Test Plan: python test/test_quantization.py
TestFxModelReportVisualizer.test_get_modules_and_features

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vspenubarthi committed:
Update on "[ao] Get feature and module names from ModelReportVisualizer class"


Summary: Added the functionality to be able to get the feature names and
module_fqns from the ModelReportVisualizer class. The purpose of this
addition is so that users can see the exact set of module_fqns or
feature names that they can filter based on, and use this information to
perform their filtering.

Test Plan: python test/test_quantization.py
TestFxModelReportVisualizer.test_get_modules_and_features

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

andrewor14 commented on torch/ao/quantization/fx/_model_report/model_report_visualizer.py:
```diff
@@ -68,7 +69,8 @@ def get_all_unique_module_fqns(self) -> Set[str]:
+        # returns the keys of the ordered dict
+        return set(self.generated_reports.keys())
 
     def get_all_unique_feature_names(self, plottable: bool) -> Set[str]:
```
nit: `plottable_features_only`? `skip_non_tensor_values`? `plottable` sounds a bit confusing. Actually why not just default this to true; I don't imagine the user will use this outside of visualization given this is part of `ModelReportVisualizer`

andrewor14 commented on torch/ao/quantization/fx/_model_report/model_report_visualizer.py:
```diff
@@ -84,7 +86,23 @@ def get_all_unique_feature_names(self, plottable: bool) -> Set[str]:
+                        unique_feature_names.add(feature_name)
+                else:
+                    # any and all features
+                    unique_feature_names.add(feature_name)
```
Can simplify this to:
```
if not plottable or type(...) == torch.Tensor:
    unique_features_names.add(feature_name)
```

vspenubarthi committed:
Update on "[ao] Get feature and module names from ModelReportVisualizer class"


Summary: Added the functionality to be able to get the feature names and
module_fqns from the ModelReportVisualizer class. The purpose of this
addition is so that users can see the exact set of module_fqns or
feature names that they can filter based on, and use this information to
perform their filtering.

Test Plan: python test/test_quantization.py
TestFxModelReportVisualizer.test_get_modules_and_features

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vspenubarthi committed:
Update on "[ao] Get feature and module names from ModelReportVisualizer class"


Summary: Added the functionality to be able to get the feature names and
module_fqns from the ModelReportVisualizer class. The purpose of this
addition is so that users can see the exact set of module_fqns or
feature names that they can filter based on, and use this information to
perform their filtering.

Test Plan: python test/test_quantization.py
TestFxModelReportVisualizer.test_get_modules_and_features

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vspenubarthi committed:
Update on "[ao] Get feature and module names from ModelReportVisualizer class"


Summary: Added the functionality to be able to get the feature names and
module_fqns from the ModelReportVisualizer class. The purpose of this
addition is so that users can see the exact set of module_fqns or
feature names that they can filter based on, and use this information to
perform their filtering.

Test Plan: python test/test_quantization.py
TestFxModelReportVisualizer.test_get_modules_and_features

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vspenubarthi commented:
@pytorchbot merge -g
